apps/rig/lib/rig/config.ex

```diff
@@ -189,8 +189,18 @@ defmodule Rig.Config do
 
   # ---
 
+  @spec resolve_path(String.t()) :: Struing.t()
```

Comment:
There's a typo. But I'd like to suggest a slightly different signature anyway - what about:

```suggestion
  @spec resolve_path(path) :: {:ok, path} | {:error, any} when path: String.t()
```

- Then, use `resolve_path/1` also in `from_file`, and rename `from_file` to `decode_json_file`.
- Instead of logging the error here, log at the call site, that is, in `check_and_update_https_config/1`. This makes the function more re-usable, since `resolve_path/1` cannot decide whether or not logging make sense (also, the fact that the log message contains "HTTPS" basically prevents any possible re-use of the function). **In any case, `check_and_update_https_config/1` should handle a non-resolvable file path (I'd suggest aborting RIG with the error message).** Note that currently, the function returns `:ok` in case the file path cannot be resolved, which is quite misleading.

Reply:
updated - please check